normalize-own-data/quantile_normalize_own_data.Rmd

```diff
@@ -0,0 +1,207 @@
+
+The submitters used RMA to process their data. 
+We have already converted the gene identifiers to Ensembl gene ids and can
+obtain the TSV file with these data from GitHub, placing it in the `data` 
+directory we create below with `bash`.
+
+```{bash eval=FALSE}
+mkdir data
+wget --directory-prefix=data https://github.com/AlexsLemonade/refinebio-examples/raw/c97cad6a8342d9bb54f120dbd7c7f12b03e3f799/compare-processing/data/GSE39842_RMA_converted.tsv
+```
+
+## Obtaining a refine.bio QN target 
+
+We'll use the terms "QN target" and "reference distribution" interchangeably to
+refer to the vector of values we use to quantile normalize `GSE39842`.
+
+We can query the refine.bio API like so:
+
+```
+https://api.refine.bio/qn_targets/?organism=<SCIENTIFIC_NAME>
+```
+
+where we would replace `<SCIENTIFIC_NAME>` with the name of the species we are
+interested in.
+
+We'd query the API for the zebrafish QN target like so:
+
+```
+https://api.refine.bio/qn_targets/?organism=DANIO_RERIO
+```
+
+Here, we'll use the `jsonlite` R package to extract the information we need to
+obtain the TSV file with the zebrafish reference distribution values. 
+
+```{r}
+qn_target_info <- jsonlite::fromJSON("https://api.refine.bio/qn_targets/?organism=DANIO_RERIO")
+```
+
+We need the `s3_url`, which tells us the link where the file is available for
+download from.
+We can read this data into R with the URL directly later on in this notebook.
+
+```{r}
+qn_url <- qn_target_info$s3_url
+```
+
+This is all the information we need to continue.
+
+```{r}
+rm(qn_target_info)
+```
+
+## Set up 
+
+We need the [`preprocessCore`](https://bioconductor.org/packages/release/bioc/html/preprocessCore.html) 
+package to perform quantile normalization. 
+
+```{r}
+if (!("preprocessCore" %in% installed.packages())) {
+  BiocManager::install("preprocessCore", update = FALSE)
+}
+```
+
+```{r}
+`%>%` <- dplyr::`%>%`
+```
+
+### Read in data
+
+First, the gene expression matrix.
+
+```{r}
+expression_file <- file.path("data", "GSE39842_RMA_converted.tsv")
+expression_df <- readr::read_tsv(expression_file)
+```
+```{r}
+expression_matrix <- expression_df %>%
+  # filter out any of the control probes
+  dplyr::filter(!grepl("AFFX", ENSEMBL)) %>%
+  as.data.frame() %>%
+  tibble::column_to_rownames("ENSEMBL") %>%
+  as.matrix()
+```
+
+Now, the QN target -- this will ultimately need to be a vector.
+
+```{r}
+qn_target <- as.vector(as.matrix(readr::read_tsv(qn_url,
+                                                 col_names = FALSE)))
+```
+
+## Quantile normalization
+
+```{r}
+qn_matrix <- preprocessCore::normalize.quantiles.use.target(expression_matrix,
+                                                            qn_target)
+```
+
+To write this to file or to do plotting with `ggplot2`, we'll want to neaten 
+things a bit (e.g., use column names).
+
+```{r}
+colnames(qn_matrix) <- colnames(expression_df)[-1]
```

Comment:
Does `preprocessCore::normalize.quantiles.use.target` not carry over the colnames?

Reply:
Correct